### PR TITLE
Fixes #92: handle the case where collection views are None

### LINKED ISSUE #92: Unexpected behaviour with adding rows to a collection
With the recent changes in https://github.com/jamalex/notion-py/pull/85, my usage of adding rows to a collection from 0.0.21 to 0.0.25 broke.

For context, I add tasks via the following: https://github.com/kevinjalbert/notion-toolbox/blob/master/alfred/src/add_task.py#L12-L23

What I end up getting is the following:

```
Traceback (most recent call last):
  File "src/add_task.py", line 28, in <module>
    row = collection.add_row()
  File "/usr/local/lib/python3.7/site-packages/notion/collection.py", line 156, in add_row
    view.set("page_sort", view.get("page_sort", []) + [row_id])
AttributeError: 'NoneType' object has no attribute 'set'
```

So for some reason, the `view` is `None` at this point.

-----

Now for the strange part. If I do the following:

```python
    collection = notion_api.tasks_database().collection

    # HACK: For some reason, we need all the views loaded up for the collection.add_row() to work
    #       it also has to be 'printed' for whatever reason, just accessing it doesn't work
    print(collection.parent.views)
...
```

It works 🤔.

I went and looked at the new changes added (https://github.com/jamalex/notion-py/blob/master/notion/collection.py#L152-L156) to see what is happening. I found that the first view for my collection's parent is `None` (which explains the exception). I went and added the following to `collection.py`:

```diff
            # make sure the new record is inserted at the end of each view
            for view in self.parent.views:
+               print(view)
+               if view is None or isinstance(view, CalendarView):
                    continue
                view.set("page_sort", view.get("page_sort", []) + [row_id])
```

It works as expected, and I saw the following output:

```
None
None
None
None
None
None
None
id='afc7fdaa-7870-48e2-9a6c-4b1c98acf05a'
```

So only the last view had a value. I'm not sure _why_ this is the case. Is there maybe some lazy loading going on for views? For additional context, I had monkey-patched out the caching a while back (https://github.com/kevinjalbert/notion-toolbox/commit/8e4e4098ea99eaa5cf24320da5805d0a341b3153) but removed that with this release.

-----

I can live with a temporary solution for my own needs. Although, I'm curious about what is happening here.

### PR DESCRIPTION
**Pull request by jamalex**
Addressing #92.

Note: includes commits from https://github.com/jamalex/notion-py/pull/95 as these were necessary in order to run smoke tests.

### CODE REVIEW

File: notion/block.py [hunk 1]
```diff
@@ -735,10 +735,23 @@ class CollectionViewBlockViews(Children):
...
 
+        i = 0
+        while view is None:
+            i += 1
+            if i > 20:
+                return None
+            time.sleep(0.1)
+            view = self._client.get_collection_view(
+                view_id, collection=self._parent.collection
+            )
```
Comment (kevinjalbert) [inline notion/block.py]: When testing, this doesn't seem to resolve the issue with the `None` views. It doesn't seem like a race-condition (I even bumped the iterations to 70). With this change, my _add_task_ takes 16~ seconds. If I remove this block of code then it takes about 2.5~ seconds (and still works).

As mentioned in my issue, when I print the `print(collection.parent.views)` that last view resolved and is a valid view. I feel like something else is up here 🤔.

For the time being, the following did resolve this for me due to the last view being valid:

```
# In collection.py L154
                if view is None or isinstance(view, CalendarView):
```

-----

I'm all on board for working with you to resolve this though.

> One guess is that perhaps there were previously some other views, and they got deleted, but their IDs were left on the CollectionViewBlock's view_ids field. So when it builds the list of views, it tries to fetch those views and doesn't find them, and hence returns None

This is something I can dig into, just need a bit more time to explore this (although the amount of returned views matches the number of views I have in my collection on Notion).

**Comment by indirectlylit** [comment]
I'm reproducing similar behavior as reported in https://github.com/jamalex/notion-py/issues/92#issuecomment-572129442 after upgrading to 0.0.25, and it persists with this PR.

Traceback:

```
Creating meeting page... Traceback (most recent call last):
  File "notion/tactical_meeting/main.py", line 156, in <module>
    main()
  File "/Users/d/Projects/le/automation/.venv/lib/python3.8/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/Users/d/Projects/le/automation/.venv/lib/python3.8/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/Users/d/Projects/le/automation/.venv/lib/python3.8/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/d/Projects/le/automation/.venv/lib/python3.8/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "notion/tactical_meeting/main.py", line 64, in main
    meeting_page = _create_meeting(client, circle_page)
  File "notion/tactical_meeting/main.py", line 20, in _create_meeting
    meeting_page = collection.add_row(
  File "/Users/d/Projects/le/notion-py/notion/collection.py", line 153, in add_row
    for view in self.parent.views:
AttributeError: 'NoneType' object has no attribute 'views'
```

I've found that I can work around this by patching

https://github.com/jamalex/notion-py/blob/56b7a904474619cf60c4768db435c921ca18f44f/notion/client.py#L87

to be

```python
    def get_block(self, url_or_id, force_refresh=True):
```

In our automation repo, you can reproduce the behavior by running:

```
python notion/tactical_meeting/main.py --circle 7c298f5274ea4a6f9cdc594ede3ccf44 --new
```

**Comment by jamalex** [comment]
Going to get this merged in so we can do testing on top of it for any remaining issues.

File: notion/block.py (hunk 1 above)
Comment (jamalex) [reply notion/block.py:751]: I've merged this (and some other fixes) to master now (not yet released, still working through the PRs and doing testing). If you're able to replicate on the latest master, let's dig in further.

**Pull request merged**